Compiler doesn't build using dotnet on windows

```
$ dotnet build FSharp.Compiler.Service.sln
Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\FSComp.fs and C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\FSComp.resx from ..\FSComp.txt since up-to-date
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\DependencyManager.fs and C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\DependencyManager.resx from ..\Microsoft.DotNet.DependencyManager\DependencyManager.txt since up-to-date
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\FSIstrings.fs and C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\FSIstrings.resx from ..\fsi\FSIstrings.txt since up-to-date
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\UtilsStrings.fs and C:\Developer\fcs-sln\artifacts\obj\FSharp.Compiler.Service\Debug\netstandard2.0\UtilsStrings.resx from ..\..\utils\UtilsStrings.txt since up-to-date
C:\Users\Eugene.Auduchinok\.nuget\packages\xlifftasks\1.0.0-beta.20420.1\build\XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named FSComp.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Service\FSharp.Compiler.Service.fsproj]

Build FAILED.

C:\Users\Eugene.Auduchinok\.nuget\packages\xlifftasks\1.0.0-beta.20420.1\build\XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named FSComp.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Service\FSharp.Compiler.Service.fsproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.75
```

It's interesting that FSharp.sln can't be built either.

```
$ dotnet build FSharp.sln
Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Server.Shared\FSharp.Compiler.Server.Shared.fsproj (in 341 ms).
  Restored C:\Developer\fcs-sln\tests\fsharpqa\testenv\src\PEVerify\PEVerify.csproj (in 341 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Service\FSharp.Compiler.Service.fsproj (in 360 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Build\FSharp.Build.fsproj (in 360 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Core.UnitTests\FSharp.Core.UnitTests.fsproj (in 389 ms).
  Restored C:\Developer\fcs-sln\tests\service\data\CSharp_Analysis\CSharp_Analysis.csproj (in 11 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Interactive.Settings\FSharp.Compiler.Interactive.Settings.fsproj (in 139 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Private.Scripting\FSharp.Compiler.Private.Scripting.fsproj (in 210 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.Private\FSharp.Compiler.Private.fsproj (in 256 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Compiler.nuget\Microsoft.FSharp.Compiler.csproj (in 270 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\fsc\fsc.fsproj (in 636 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\fsiAnyCpu\fsiAnyCpu.fsproj (in 78 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Compiler.Private.Scripting.UnitTests\FSharp.Compiler.Private.Scripting.UnitTests.fsproj (in 371 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.DependencyManager.Nuget\FSharp.DependencyManager.Nuget.fsproj (in 130 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\FSharp.Core\FSharp.Core.fsproj (in 94 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Compiler.Service.Tests\FSharp.Compiler.Service.Tests.fsproj (in 281 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Compiler.UnitTests\FSharp.Compiler.UnitTests.fsproj (in 368 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\fsi\fsi.fsproj (in 443 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Compiler.ComponentTests\FSharp.Compiler.ComponentTests.fsproj (in 548 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Build.UnitTests\FSharp.Build.UnitTests.fsproj (in 205 ms).
  Restored C:\Developer\fcs-sln\src\fsharp\Microsoft.DotNet.DependencyManager\Microsoft.DotNet.DependencyManager.fsproj (in 135 ms).
  Restored C:\Developer\fcs-sln\tests\FSharp.Test.Utilities\FSharp.Test.Utilities.fsproj (in 1.36 sec).
  Restored C:\Developer\fcs-sln\tests\fsharp\FSharpSuite.Tests.fsproj (in 1.68 sec).
C:\Developer\fcs-sln\artifacts\Bootstrap\fsc\Microsoft.FSharp.Targets(281,9): error MSB6006: "dotnet.exe" exited with code -2147450749. [C:\Developer\fcs-sln\src\fsharp\FSharp.Core\FSharp.Core.fsproj]
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\Microsoft.DotNet.DependencyManager\Debug\net472\DependencyManager.fs and C:\Developer\fcs-sln\artifacts\obj\Microsoft.DotNet.DependencyManager\Debug\net472\DependencyManager.resx from DependencyManager.txt since up-to-date
FSharpEmbedResourceText: Skipping generation of C:\Developer\fcs-sln\artifacts\obj\Microsoft.DotNet.DependencyManager\Debug\net472\UtilsStrings.fs and C:\Developer\fcs-sln\artifacts\obj\Microsoft.DotNet.DependencyManager\Debug\net472\UtilsStrings.resx from ..\..\utils\UtilsStrings.txt since up-to-date
  PEVerify -> C:\Developer\fcs-sln\artifacts\bin\PEVerify\Debug\net472\PEVerify.exe
  PEVerify -> C:\Developer\fcs-sln\artifacts\bin\PEVerify\Debug\netcoreapp3.1\PEVerify.dll

Build FAILED.

C:\Developer\fcs-sln\artifacts\Bootstrap\fsc\Microsoft.FSharp.Targets(281,9): error MSB6006: "dotnet.exe" exited with code -2147450749. [C:\Developer\fcs-sln\src\fsharp\FSharp.Core\FSharp.Core.fsproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.96
```

Is there any workaround to make `dotnet build` work?

Just adding in that I can reproduce this issue on my macos environment as well.  I updated main yesterday and running build.sh now gives three classes of errors:

* duplicate assembly attributes in PEVerify.csproj
* XliffTasks duplicate resource issues in the DependencyManager infrastructure
* Language/parser changes to numeric literals breaking the FSharp.Core.UnitTests project

```
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/.NETCoreApp,Version=v3.1.AssemblyAttributes.cs(4,12): error CS0579: Duplicate 'global::System.Runtime.Versioning.TargetFrameworkAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(14,12): error CS0579: Duplicate 'System.Reflection.AssemblyCompanyAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(15,12): error CS0579: Duplicate 'System.Reflection.AssemblyConfigurationAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(17,12): error CS0579: Duplicate 'System.Reflection.AssemblyFileVersionAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(18,12): error CS0579: Duplicate 'System.Reflection.AssemblyInformationalVersionAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(19,12): error CS0579: Duplicate 'System.Reflection.AssemblyProductAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(20,12): error CS0579: Duplicate 'System.Reflection.AssemblyTitleAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/oss/fsharp/artifacts/obj/PEVerify/Debug/netcoreapp3.1/PEVerify.AssemblyInfo.cs(21,12): error CS0579: Duplicate 'System.Reflection.AssemblyVersionAttribute' attribute [/Users/chethusk/oss/fsharp/tests/fsharpqa/testenv/src/PEVerify/PEVerify.csproj]
/Users/chethusk/.nuget/packages/xlifftasks/1.0.0-beta.20502.2/build/XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named FSInteractiveSettings.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [/Users/chethusk/oss/fsharp/src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj]
/Users/chethusk/.nuget/packages/xlifftasks/1.0.0-beta.20502.2/build/XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named FSComp.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [/Users/chethusk/oss/fsharp/src/fsharp/FSharp.Compiler.Service/FSharp.Compiler.Service.fsproj]
/Users/chethusk/.nuget/packages/xlifftasks/1.0.0-beta.20502.2/build/XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named DependencyManager.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [/Users/chethusk/oss/fsharp/src/fsharp/Microsoft.DotNet.DependencyManager/Microsoft.DotNet.DependencyManager.fsproj]
/Users/chethusk/.nuget/packages/xlifftasks/1.0.0-beta.20502.2/build/XliffTasks.targets(123,5): error : Two or more source files to be translated in the same project are named FSDependencyManager.txt. Give them unique names or set unique XlfTranslatedFilename metadata. [/Users/chethusk/oss/fsharp/src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Nuget.fsproj]
/Users/chethusk/oss/fsharp/tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModule2.fs(380,42): error FS1151: This number is outside the allowable range for signed native integers [/Users/chethusk/oss/fsharp/tests/FSharp.Core.UnitTests/FSharp.Core.UnitTests.fsproj]
/Users/chethusk/oss/fsharp/tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModule2.fs(383,61): error FS1151: This number is outside the allowable range for signed native integers [/Users/chethusk/oss/fsharp/tests/FSharp.Core.UnitTests/FSharp.Core.UnitTests.fsproj]
```

I'm working on getting a binary log to submit here, but first have to scrub all my sensitive environment variables from it.

--- a/FSharpBuild.Directory.Build.props
+++ b/FSharpBuild.Directory.Build.props
@@ -8,8 +8,9 @@
   </PropertyGroup>
 
   <!-- directory locations -->
-  <PropertyGroup>
-    <DisableAutoSetFscCompilerPath Condition="'$(DisableAutoSetFscCompilerPath)' == '' and '$(Configuration)' != 'Proto'">true</DisableAutoSetFscCompilerPath>
+
+  <PropertyGroup Condition="'$(OS)' != 'Unix' and '$(DisableAutoSetFscCompilerPath)' != 'true' and '$(Configuration)' != 'Proto' and '$(DOTNET_HOST_PATH)' != ''">
+    <DisableAutoSetFscCompilerPath>true</DisableAutoSetFscCompilerPath>
   </PropertyGroup>
 
   <PropertyGroup>

--- a/FSharpBuild.Directory.Build.props
+++ b/FSharpBuild.Directory.Build.props
@@ -9,6 +9,10 @@
 
   <!-- directory locations -->
   <PropertyGroup>
+    <DisableAutoSetFscCompilerPath Condition="'$(DisableAutoSetFscCompilerPath)' == '' and '$(Configuration)' != 'Proto'">true</DisableAutoSetFscCompilerPath>
+  </PropertyGroup>
+
+  <PropertyGroup>
     <FSharpSourcesRoot>$(RepoRoot)src</FSharpSourcesRoot>
     <FSharpTestsRoot>$(RepoRoot)tests</FSharpTestsRoot>
     <SymStoreDirectory>$(ArtifactsDir)\SymStore</SymStoreDirectory>
@@ -16,7 +20,7 @@
     <ValueTupleImplicitPackageVersion>4.4.0</ValueTupleImplicitPackageVersion>
     <WarningsAsErrors>1182;0025;$(WarningsAsErrors)</WarningsAsErrors>
     <OtherFlags>$(OtherFlags) --nowarn:3384</OtherFlags>
-</PropertyGroup>
+  </PropertyGroup>
 
   <!-- nuget -->
   <PropertyGroup>

